### ISSUE TITLE
PNPM is (still?) not correctly identifying the node version

### ISSUE THREAD

**Issue by arkanes** [pnpm/pnpm#5266]
### pnpm version:
7.9.5

### Code to reproduce the issue:
pnpm install

### Expected behavior:
installation proceeds as normal

### Actual behavior:
 ERR_PNPM_UNSUPPORTED_ENGINE  Unsupported environment (bad pnpm and/or Node.js version)

### Additional information:

 - `node -v` prints:
 - v18.8.0
 - Windows, macOS, or Linux?:
Windows 10

This is the same issue as #4203 and was either never fixed or has regressed.

This can be reproduced by downloading the 7.9.5 version from github releases and running pnpm install twice. See attached console log

```
D:\dev
λ mkdir repro

    Directory: D:\dev

Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
d----           8/25/2022  5:19 PM                repro

D:\dev
λ cd repro
D:\dev\repro
λ Invoke-WebRequest https://github.com/pnpm/pnpm/releases/download/v7.9.5/pnpm-win-x64.exe -OutFile pnpm-win-x64.exe
D:\dev\repro
λ .\pnpm-win-x64.exe --version
7.9.5
D:\dev\repro
λ .\pnpm-win-x64.exe create svelte@latest test
../../.pnpm-store/v3/tmp/dlx-18352       |   +5 +
Packages are hard linked from the content-addressable store to the virtual store.
  Content-addressable store is at: D:\.pnpm-store\v3
  Virtual store is at:             ../../.pnpm-store/v3/tmp/dlx-18352/node_modules/.pnpm
../../.pnpm-store/v3/tmp/dlx-18352       | Progress: resolved 5, reused 5, downloaded 0, added 5, done

create-svelte version 2.0.0-next.162

Welcome to SvelteKit!

This is beta software; expect bugs and missing features.

Problems? Open an issue on https://github.com/sveltejs/kit/issues if none exists already.

√ Which Svelte app template? » SvelteKit demo app
√ Add type checking with TypeScript? » Yes, using JavaScript with JSDoc comments
√ Add ESLint for code linting? ... No / Yes
√ Add Prettier for code formatting? ... No / Yes
√ Add Playwright for browser testing? ... No / Yes

Your project is ready!
✔ Type-checked JavaScript
  https://www.typescriptlang.org/tsconfig#checkJs

Install community-maintained integrations:
  https://github.com/svelte-add/svelte-adders

Next steps:
  1: cd test
  2: npm install (or pnpm install, etc)
  3: git init && git add -A && git commit -m "Initial commit" (optional)
  4: npm run dev -- --open

To close the dev server, hit Ctrl-C

Stuck? Visit us at https://svelte.dev/chat

D:\dev\repro
λ cd test
D:\dev\repro\test
λ pnpm install
Packages: +169
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
Packages are hard linked from the content-addressable store to the virtual store.
  Content-addressable store is at: D:\.pnpm-store\v3
  Virtual store is at:             node_modules/.pnpm
Progress: resolved 190, reused 169, downloaded 0, added 169, done

dependencies:
+ @fontsource/fira-mono 4.5.9
+ cookie 0.4.2 (0.5.0 is available)

devDependencies:
+ @sveltejs/adapter-auto 1.0.0-next.66
+ @sveltejs/kit 1.0.0-next.442
+ @types/cookie 0.5.1
+ svelte 3.49.0
+ svelte-check 2.8.1
+ typescript 4.7.4
+ vite 3.0.9
D:\dev\repro\test
λ pnpm install
Lockfile is up to date, resolution step is skipped
 ERR_PNPM_UNSUPPORTED_ENGINE  Unsupported environment (bad pnpm and/or Node.js version)

Your Node version is incompatible with "/@sveltejs/kit/1.0.0-next.442_svelte@3.49.0+vite@3.0.9".

Expected version: >=16.9
Got: v14.19.2

This is happening because the package's manifest has an engines.node field specified.
To fix this issue, install the required Node version.
D:\dev\repro\test
λ node -v
v18.8.0

**Comment by beeb** [comment]
Experiencing the same on macOS and pnpm 7.9.5.

Regardless of my `env use --global` setting, it's always using v14.x

**Comment by perfStack** [comment]
Experiencing the same on Arch linux and pnpm 7.9.5.

Steps to fix :-
1) Created .npmrc with  `use-node-version=18.8.0`
2) ran pnpm i

output:
```
Fetching Node.js 18.8.0 ...
Node.js 18.8.0 is activated
  $InstallDir/pnpm/node -> /home/$User/.local/share/pnpm/nodejs/18.8.0/bin/node
```
Fixed the issue for me.

**Comment by AdriVanHoudt** [comment]
I also had this issue

* pnpm version 7.14.2 (installed via homebrew on macOS 13, but updated over time with `pnpm add -g pnpm`)
* run `pnpm env --global 18`
* `node -v` returns 18
* commands ran via pnpm give engine warnings
* but things like `pnpm node` run the correct node version but still print the warning

Fixed the issue by running `pnpm add -g pnpm`

**Comment by pete-willard** [comment]
The same happened to me on pnpm v7.26.0. I am on Manjaro Linux. I must have had node v14.21.2 installed via nvm at some point.

**Comment by jcc1997** [comment]
The same happened in windows on pnpm v7.26.0. And can't run pnpm add -g pnpm.

**Comment by ZYinMD** [comment]
Same problem on Windows and can't be solved by `pnpm add -g pnpm` because #5700.

The suggested workaround to update pnpm in #5700 is `iwr https://get.pnpm.io/install.ps1 -useb | iex`, but that doesn't solve this node version issue.

**Comment by brxken128** [comment]
Running Gentoo, had the issue with both the repository node (`v18.14.0`) and `fnm`'s `v19.7.0` (with pnpm `v7.28.0`).

Running `pnpm add -g pnpm` fixed the issue.

**Comment by mllmei** [comment]
> I also had this issue
> 
> * pnpm version 7.14.2 (installed via homebrew on macOS 13, but updated over time with `pnpm add -g pnpm`)
> * run `pnpm env --global 18`
> * `node -v` returns 18
> * commands ran via pnpm give engine warnings
> * but things like `pnpm node` run the correct node version but still print the warning
> 
> Fixed the issue by running `pnpm add -g pnpm`

This is the only correct answer which finally worked in my project

**Comment by reikrom** [comment]
`macOs Ventura 13.2.1`
Using Volta to switch node versions between projects. `14` `16` and `18`
`pnpm` seems to be stuck on the node version it was installed globally at the time. 
Reinstalling pnpm `-g` while having the desired version works as a hack.

**Comment by brxken128** [comment]
> `macOs Ventura 13.2.1`
> Using Volta to switch node versions between projects. `14` `16` and `18`
> `pnpm` seems to be stuck on the node version it was installed globally at the time. 
> Reinstalling pnpm `-g` while having the desired version works as a hack.

I doubt this is the case, as I have _never_ installed Node v14.19.2 (or a version close). That, and it seems as though a lot of people have experienced this issue randomly - it's extremely unlikely that everyone had _**exactly**_ Node v14.19.2 when they installed `pnpm`, IMO.

**Comment by SAMDevelopment** [comment]
I had the same issue when I had pnpm installed through volta. After uninstalling it and following the regular pnpm install from the website it worked again.

**Comment by bradyab** [comment]
none of these solutions worked for me, I ended up simply installing the right node version on my machine with `n <version>`

**Comment by kyriacos** [comment]
I had to do a number of things to get this working. Disabled `nodenv` from loading and just use the `pnpm` global version of node. And i also had to add the specific version in my `.npmrc` file => `use-node-version=18.15.0`.

Also set in the `.node-version` for the project and updated the `engines` field. Once i did that an ran `pnpm install -g pnpm` it finally started working okay.

I was working in a monorepo and the problem was only for packages that did not have an `.npmrc` or `.node-version` when running a `-r` command like install.

Hope that helps. Thanks for all the suggestions my missing piece was the npmrc change.

pnpm v8.1.1

**Comment by theoephraim** [comment]
Also running into something like this when trying to have pnpm coexist with volta.

While having pnpm manage the node version is a noble mission, it seems that the existence of the (what should be optional) feature is somehow making things _more_ complex instead of less... :(

More details:

Running `which node` results in something like `/Users/theo/.volta/bin/node` (which lets volta manage the version).
But adding `"whichnode": "which node"` to package.json scripts and running `pnpm run whichnode`) results in something like `/Users/theo/.volta/tools/image/node/18.16.0/bin/node`

While adding a specific node version in .npmrc file does tell pnpm which pnpm-managed version to use, I do not think this should be a requirement. From the behaviour above, it seems that pnpm is trying to fallback to whatever version is running locally, and while that seems reasonable, it seems to be somehow resolving the version that is running during install.

(sidenote - re-running `volta install pnpm` seems to at least reset it, the same way a `pnpm install -g pnpm` would, but keeping things in volta land)

**Comment by TurboTobias** [comment]
Helper that force updates the node version by `$ env 18.16.0`

```sh
env() {
  echo "will set node and pnpm to v$1"
  echo "use-node-version=$1" >> .npmrc
  nvm use "$1"
  pnpm env use --global "$1"
  pnpm add -g pnpm
}
```

**Comment by brxken128** [comment]
> Helper that force updates the node version by `$ env 18.16.0`
> 
> ```sh
> env() {
>   echo "will set node and pnpm to v$1"
>   echo "use-node-version=$1" >> .npmrc
>   nvm use "$1"
>   pnpm env use --global "$1"
>   pnpm add -g pnpm
> }
> ```

Not everyone uses `nvm`, and it's not really needed for this task AFAIK.

`pnpm add -g pnpm does the trick.

**Comment by geiszla** [comment]
I'm trying to deploy to GCloud App Engine and getting the same issue. Neither of these proposed solutions would work there, would they? Has someone been able to solve this without adding `engine-strict=false` to `.npmrc`?

**Comment by FrederickEngelhardt** [comment]
Noting I had this happen. I originally installed pnpm on node 16 then jumped to node 18 with `fnm` I still saw a `node-version=16.14.0` when running `pnpm config list`

**Comment by vjpr** [comment]
What does the following output: `cat $(command which pnpm)`

**Comment by FrederickEngelhardt** [comment]
The bin file
```javascript
#!/usr/bin/env node
const [major, minor] = process.version.slice(1).split('.')
const COMPATIBILITY_PAGE = `Visit https://r.pnpm.io/comp to see the list of past pnpm versions with respective Node.js version support.`

// We don't use the semver library here because:
//  1. it is already bundled to dist/pnpm.cjs, so we would load it twice
//  2. we want this file to support potentially older Node.js versions than what semver supports
if (major < 16 || major == 16 && minor < 14) {
  console.log(`ERROR: This version of pnpm requires at least Node.js v16.14
The current version of Node.js is ${process.version}
${COMPATIBILITY_PAGE}`)
  process.exit(1)
}

global['pnpm__startedAt'] = Date.now()
require('../dist/pnpm.cjs')

// if you want to debug at your local env, you can use this
// require('../lib/pnpm')
```

**Comment by vjpr** [comment]
Modify that file and add `which node` to see where it is locating node.

pnpm usually runs from a bin stub for me...

**Comment by FrederickEngelhardt** [comment]
And script to check this real quick when upgrading
```
fnm uninstall 16.20.1;
fnm uninstall 18.17.0;

fnm install 16.20.1; fnm use 16.20.1; node --version; npm i -g pnpm; which pnpm; 
echo "console.log(\"NODE_VERSION\", process.version);" >> $(command which pnpm)
pnpm;

fnm install 18.17.0; fnm use 18.17.0; node --version; npm i -g pnpm; which pnpm; 
echo "console.log(\"NODE_VERSION\", process.version);" >> $(command which pnpm)
pnpm;

fnm use 16.20.1;
pnpm;
```

You will probably want to clean up those pnpm packages so either uninstalling node or removing `pnpm` from each version will remove that extract log now added to the binary file.

Outputs
```
/Users/<USERNAME>/Library/Caches/fnm_multishells/10299_1682566412504/bin/pnpm
NODE_VERSION v16.20.1

/Users/<USERNAME>/Library/Caches/fnm_multishells/10299_1682566412504/bin/pnpm
NODE_VERSION v18.17.0

NODE_VERSION v16.20.1
```

Both should output the NODE_VERSION on the top. Though for me I think I was messing around with `pnpm env` and I got the node-version added to my config that way.

**Comment by vjpr** [comment]
I don't see a problem with that output...

**Comment by moltar** [comment]
There's also an issue running inside GitHub Workflow. It seems like `pnpm` is not recognizing that the Node version already exists and updated.

We have `actions/setup-node@v3` configured to set the exact version.

But then pnpm tries to update again for some reason.

![screenshot-20230927T122122-Uu0pjTEV@2x](https://github.com/pnpm/pnpm/assets/491247/4fee5a3e-564f-46ad-82bd-1fdf7d8bc8dd)

**Comment by biro456** [comment]
I'm seeing this problem while using PNPM `8.8.0` while deploying to Google App Engine with Node 20 runtime.

I see no mention of the older Node version being activated by PNPM in the logs, but it still is resolving to the wrong version.
Perhaps an interesting thing is that both my workspace and a dependency have `engines.node` constraints (`~20` and `>=20` respectively), but only the dependency one is causing a failure.

As a workaround, I've disabled the `engine-strict=true` I have in the repo's `.npmrc` and the installation completed.

<details>
  <summary>Logs</summary>

```
===> BUILDING
Timer: Builder started at 2023-09-28T05:32:33Z
=== Node.js - Runtime (google.nodejs.runtime@1.0.0) ===
Using runtime version from GOOGLE_RUNTIME_VERSION: 20.6.0
***** CACHE MISS: "nodejs"
Installing Node.js v20.6.0.
2023/09/28 05:32:33 [DEBUG] GET https://dl.google.com/runtimes/ubuntu2204/nodejs/nodejs-20.6.0.tar.gz
=== Node.js - Pnpm (google.nodejs.pnpm@0.1.0) ===
2023/09/28 05:32:37 [DEBUG] GET https://registry.npmjs.org/pnpm
***** CACHE MISS: "pnpm_engine"
Installing pnpm v8.8.0
2023/09/28 05:32:37 [DEBUG] GET https://github.com/pnpm/pnpm/releases/download/v8.8.0/pnpm-linux-x64
Setting environment variable PATH=/layers/google.nodejs.pnpm/pnpm_engine/bin:/layers/google.nodejs.runtime/node/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
--------------------------------------------------------------------------------
Running "pnpm install (CI=true NODE_ENV=development)"
Lockfile is up to date, resolution step is skipped
 ERR_PNPM_UNSUPPORTED_ENGINE  Unsupported environment (bad pnpm and/or Node.js version)

Your Node version is incompatible with "/@biro456/framework/0.1.2".

Expected version: >=20
Got: v18.5.0

This is happening because the package's manifest has an engines.node field specified.
To fix this issue, install the required Node version.
Done "pnpm install (CI=true NODE_ENV=development)" (491.23125ms)
```
</details>

**Comment by paro-paro** [comment]
I believe this is still an issue on pnpm `8.10.5` when installed as a standalone.

Running `pnpm install` with `engine-strict=true` set in `.npmrc` resolves to a "random" node version.

**Note:** running `pnpm add -g pnpm` after `pnpm env use` or setting the `use-node-version` option in `.npmrc` "fixes" the issue but I guess it should work without workarounds.

<img width="746" alt="Screenshot 2023-11-21 at 12 29 24" src="https://github.com/pnpm/pnpm/assets/43136874/55fd47ff-1e6a-4362-9bee-a651b6ccd181">

**Comment by lirbank** [comment]
Thanks so much @biro456 and @paro-paro! I had this exact same issue I've uninstalled nvm, pnpm, all versions of node from all package managers and it still didn't work.

Then I commented out `engine-strict=true` in my `.npmrc` file and everything works like a charm.

I am on pnpm 8.11.0

The error I get when having `engine-strict=true` enabled is:
```
Expected version: >=18.17.0
Got: v18.5.0
```

The only Node version I have currently is 20.10.0, and it is installed with `pnpm env use --global 20`.

```sh
$ pnpm env list
  20.10.0
```

**Comment by CaptainOfPhB** [comment]
> Modify that file and add `which node` to see where it is locating node.
> 
> pnpm usually runs from a bin stub shell script for me...I found that earlier versions hard-coded the node version.
> 
> You should also check if you use `pnpm env`.
> 
> This script is usually at `/Users/Vaughan/Library/pnpm/pnpm`. If `/Users/Vaughan/Library/pnpm/node` exists, it will use that.

The answer from @vjpr works for me. I use `n` to manage my node version, but I got a wrong node version and location after I installed a new node version:

```bash
$ sudo n lts
     copying : node/20.10.0
   installed : v20.10.0 to /usr/local/bin/node
      active : v18.8.0 at /Users/captainofphb/Library/pnpm/node

$ node -v
v18.8.0

$ which node
/Users/captainofphb/Library/pnpm/node
```

After I removed `pnpm env`:

```bash
$ pnpm env list
18.8.0

$ pnpm env remove --global 18.8.0
Node.js 18.8.0 was removed
/Users/captainofphb/Library/pnpm/nodejs/18.8.0

$ node -v
v20.10.0

$ which node
/usr/local/bin/node
```

Good!

**Comment by stolinski** [comment]
<img width="379" alt="Screenshot 2024-01-10 at 6 44 18 PM" src="https://github.com/pnpm/pnpm/assets/669383/b41af166-b859-467f-b4e6-7f0f817af973">
Just another data point here. I'm also having this issue. On the latest pnpm. I don't use n, nvm or any other node manager. 18.5 does not exist on my system. I have no idea where pnpm is pulling that version from.

**Comment by lewinskimaciej** [comment]
Have exact same issue:
```
C:\(...) (develop -> origin)
λ node -v
v20.11.0

C:\(...) (develop -> origin)
λ pnpm install
Scope: all 11 workspace projects
Lockfile is up to date, resolution step is skipped
 ERR_PNPM_UNSUPPORTED_ENGINE  Unsupported environment (bad pnpm and/or Node.js version)

Your Node version is incompatible with "/@metamask/eth-sig-util/7.0.1".

Expected version: ^16.20 || ^18.16 || >=20
Got: v18.5.0

This is happening because the package's manifest has an engines.node field specified.
To fix this issue, install the required Node version.

```
running on Windows if that matters, using nvm-windows but this also happened when i installed node manually. Path to node is set properly and node is usable everywhere besides pnpm.

**Comment by lewinskimaciej** [comment]
> i had this issue some time ago and the reason was ( in my case ) that somehow I had the $PATH variable containing the entry of different node installation rather than the nvm I was using. the solution wasn't obvious so it took a while to find out.
> 
> my take on this is that if you use pnpm to manage node version AND something else, don't. Use only one. I chose nvm, not perfectly happy with it but it does the job

It's not that easy. pnpm didn't work with just `pnpm env` for me either(tried it before installing nvm-windows), it also lists wrong version. I tried every combination and on Windows it just doesn't recognize the version for some reason. Environment variable properly leads to the C:/Program Files/nodejs which is a symlink made by nvm, literally everything else works, just not pnpm and I'm not sure why pnpm just doesnt try to get node version by calling node -v, instead of reading something manually somewhere? For now I just disabled `engine-strict` in our project in `.npmrc` but that's hardly a solution.

**Comment by karimfromjordan** [comment]
> <img alt="Screenshot 2024-01-10 at 6 44 18 PM" width="379" src="https://private-user-images.githubusercontent.com/669383/295766188-b41af166-b859-467f-b4e6-7f0f817af973.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU0Mjc1MTEsIm5iZiI6MTcwNTQyNzIxMSwicGF0aCI6Ii82NjkzODMvMjk1NzY2MTg4LWI0MWFmMTY2LWI4NTktNDY3Zi1iNGU2LTdmMGY4MTdhZjk3My5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDExNlQxNzQ2NTFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kMzgyZmY3YmM5OWZhOTQ4NGU2Y2MzNzAwYzMzMDE4ZTU5ZGNhMWNiMGQ2MDIyZWZkNTcxYTFlZWUxYzFlNTg4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Dg2CSdpC7DP4UrhtuX6fojeNfGKSUO78YlO6rBfraTQ"> Just another data point here. I'm also having this issue. On the latest pnpm. I don't use n, nvm or any other node manager. 18.5 does not exist on my system. I have no idea where pnpm is pulling that version from.

I'm experiencing the exact same thing.

**Comment by CaptainCodeman** [comment]
I had this issue, it was caused by multiple node versions existing and vs.code using a different one to the pnpm managed version.

Add a script to package.json to run `which node` and then compare it to what you see in the vs.code terminal and regular terminal, remove any non-pnpm managed versions if they show up.

**Comment by magnusriga** [comment]
@zkochan Could you please explain where pnpm searches for the version of node that it uses? I have the same issue as others, i.e. that pnpm expected a certain version (the one I listed in `engines`), but instead "got" an entirely different (and old) version. As far as I know, I do not have that old version installed.

PS 1: I am developing in a vscode devcontainer where node was installed via an nvm script, then pnpm installed via curl. `node -v` and `pnpm -v` both print the latest versions.

PS 2: I checked out my PATH variable, and it does not contain that old outdated node version which pnpm seems to use.

If we knew where pnpm found its node version, then perhaps we could solve this quickly.

**Comment by kaechele** [comment]
Hit the same issue today. Only had time to look into possible root-cause briefly.

My guess is that the culprit is https://github.com/pnpm/pnpm/blob/1a3449e94065543ab97d015f220727950875135d/config/package-is-installable/src/getSystemNodeVersion.ts#L6
The release binaries seemingly bundle nodejs 18.5.0 [^1] as their runtime to bootstrap pnpm by means of the [deprecated `@vercel/pkg` packager](https://github.com/vercel/pkg?tab=readme-ov-file#pkg). The above mentioned check may be failing (might be worth trying to check for `process.pkg !== undefined` instead?) and the version being returned is the bundled nodejs version 18.5.0, not the system node version in the your environment's `PATH`.

This is why the issue goes away when you install via `npm -g` instead of piping curl scripts to bash or using the GitHub Release provided binaries. pnpm doesn't need the bundled nodejs in that case and will always simply return the node version that was used to invoke pnpm.

[^1]: As seen by running `strings ~/.local/share/pnpm/pnpm | grep "18.5.0"`

**Comment by antony** [comment]
I believe I'm still seeing this issue too, on github actions which use node 18.19 by default, it seems.

<img width="833" alt="Screenshot 2024-02-03 at 16 07 43" src="https://github.com/pnpm/pnpm/assets/218949/39fac84f-874b-4ea7-8987-3d43e2b594b7">

As you can see, I run `pnpm exec node -v` and the version is correct as set in my `.npmrc`

```
use-node-version=20.11.0
node-version=20.11.0
engine-strict=true
```

when it runs `pnpm install`, for some reason, it's using the version of node which github actions comes with - 18.19

**Comment by antony** [comment]
In case it helps, here's [a reproduction](https://github.com/antony/pnpm-5266-repro/actions/runs/7768314875/job/21186098819) on github actions.

**Comment by lirbank** [comment]
I'm now on pnpm 8.15.2

It's been working for a while, but I recently added Next.js to our monorepo and the issue reappeared.

```sh
% pnpm install 
Scope: all 3 workspace projects
Lockfile is up to date, resolution step is skipped
 ERR_PNPM_UNSUPPORTED_ENGINE  Unsupported environment (bad pnpm and/or Node.js version)

Your Node version is incompatible with "/next/14.1.0(react-dom@18.2.0)(react@18.2.0)".

Expected version: >=18.17.0
Got: v18.5.0

This is happening because the package's manifest has an engines.node field specified.
To fix this issue, install the required Node version.


% which node
~/.nvm/versions/node/v20.11.0/bin/node


% node -v   
v20.11.0
```

**Comment by DaLukasDev** [comment]
FWIW I'm running into the same issue when using the install script from the installation section. 

If PNPM is installed through the install.sh and I run `pnpm config list` I will get 
`user-agent=pnpm/8.15.3 npm/? node/v18.5.0 darwin arm64`

However if I install pnpm through homebrew it will infer my node version based on what version nvm is on.
`user-agent=pnpm/8.15.3 npm/? node/v20.9.0 darwin arm64`

I do not have a version that is even remotly close to 18.5.0 installed on my system and I also have no idea where pnpm is pulling that from.
The issue started just now and hasn't been an issue in the past.

OS: Mac OS Sonoma (14.3.1)
Node version manager: NVM
using node version 20.9.0

tldr; Try installing it through homebrew if you're using macos.

**Comment by silasabbott** [comment]
I want to point out that many of us are getting the same exact Node version mentioned by pnpm:

```
Got: v18.5.0
```

Earlier versions of pnpm seem to be "locked" to an exact 14.x.x version if you look at comments further back.

I'm on macOS and I'm using `pnpm env` to manage node versions right now. A full pnpm reinstall didn't fix it for me. `pnpm add -g pnpm` is also not a correct fix because now pnpm recommends that you install through the convenience script.
